boards/arm/pinetime_devkit0/CMakeLists.txt

```diff
@@ -2,4 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
-zephyr_library_sources(key_out.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO
+  key_out.c
+  )
```

Comment:
this will result in an empty library if `!CONFIG_GPIO`

Reply:
@gmarull I am not sure this is what you meant, but I moved the if-statement, so that `zephyr_library()` is only called when there is a source file.